### PR TITLE
Lab2 ff fixes the bad average reward problem

### PR DESCRIPTION
**Pull request by rlindsberg**
Skærmbillede 2020-12-30 kl  01 33 05

Skærmbillede 2020-12-30 kl  01 34 38

### CODE REVIEW

File: lab2/problem1/DQN_problem.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def running_average(x, N):
 
 # fill buffer with random exps
 state = env.reset()
-for i in range(10000):
+for i in range(100):
```
Comment (rlindsberg) [inline lab2/problem1/DQN_problem.py:62]: Det här spelar inte så mycket roll, 100 eller 10000. 100 tar så mycket mindre tid.

Comment (rlindsberg) [inline lab2/problem1/DQN_problem.py:62]: Det här spelar inte så mycket roll, 100 eller 10000. 100 tar så mycket mindre tid.

File: lab2/problem1/DQN_problem.py [hunk 2]
```diff
@@ -78,6 +78,12 @@ def running_average(x, N):
     total_episode_reward = 0.
     t = 0
     epsilon = np.maximum(min_eps, max_eps - ((max_eps - min_eps) * i) / ((0.9 * N_episodes) - 1), casting='same_kind')
+
+    if i % 100 == 0:
+        print("\n Epsilon is ")
+        print(epsilon)
```
Comment (rlindsberg) [inline lab2/problem1/DQN_problem.py:84]: Den skrivs ut för varje episode efter 458 såna, vet inte varför...

File: lab2/problem1/DQN_problem.py [hunk 3]
```diff
@@ -118,6 +124,12 @@ def running_average(x, N):
             running_average(episode_reward_list, n_ep_running_average)[-1],
             running_average(episode_number_of_steps, n_ep_running_average)[-1]))
 
+    ave_score = running_average(episode_reward_list, n_ep_running_average)[-1]
+    if ave_score != 0:
+        if ave_score > 100:
+            print("i is" + str(i) + '\n')
+            torch.save(agent.q_network_local.state_dict(), 'neural-network-1.pth')
```
Comment (rlindsberg) [inline lab2/problem1/DQN_problem.py:131]: Man kan då avbryta programmet här

**Review by mipalme — APPROVED** [review]
